From 01d7316f6773591186f606de5e3b12de28bb0cd9 Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Sun, 9 Jul 2023 18:45:15 -0600
Subject: [PATCH 4/5] hcrypto: rename abort to _afscrypto_abort
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The Linux 6.5 commit:
    panic: make function declarations visible (d9cdb43189)
added a declaration for abort into panic.h.

When building the Linux kernel module, the build fails with the
following:

  src/crypto/hcrypto/kernel/config.h:95:20: error: static declaration of
      ‘abort’ follows non-static declaration
    95 | static_inline void abort(void) {osi_Panic("hckernel aborting\n"
         );}
       |                    ^~~~~
    ...
                 from ./include/linux/wait.h:9,
                 from /openafs/src/afs/sysincludes.h:118,
                 from /openafs/src/crypto/hcrypto/kernel/config.h:30:
    ./include/linux/panic.h:36:6: note: previous declaration of ‘abort’
         with type ‘void(void)’
    36 | void abort(void);
       |      ^~~~~

Update the declaration in hcrypto/kernel/config.h to change the function
name from abort to _afscrypto_abort and use a preprocessor define to
map abort to _afscrypto_abort.

Reviewed-on: https://gerrit.openafs.org/15501
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
Tested-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit c4c16890d9d2829f6bef1ef58feafb30b1d59da3)

Change-Id: I54cc9156b98320d04fe6f7bb595a150d5ba87b49
Reviewed-on: https://gerrit.openafs.org/15523
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Reviewed-by: Mark Vitale <mvitale@sinenomine.net>
Reviewed-by: Stephan Wiesand <stephan.wiesand@desy.de>
(cherry picked from commit 538f450033a67e251b473ff92238b3124b85fc72)
---
 src/crypto/hcrypto/kernel/config.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/crypto/hcrypto/kernel/config.h b/src/crypto/hcrypto/kernel/config.h
index 9623fa9e7..ea0f60b64 100644
--- a/src/crypto/hcrypto/kernel/config.h
+++ b/src/crypto/hcrypto/kernel/config.h
@@ -91,7 +91,13 @@ extern int osi_readRandom(void *, afs_size_t);
 static_inline pid_t getpid(void) {return 1;};
 #endif
 static_inline int open(const char *path, int flags, ...) {return -1;}
-static_inline void abort(void) {osi_Panic("hckernel aborting\n");}
+
+#ifdef abort
+# undef abort
+#endif
+#define abort _afscrypto_abort
+static_inline void _afscrypto_abort(void) {osi_Panic("hckernel aborting\n");}
+
 static_inline void rk_cloexec(int fd) {}
 static_inline ssize_t read(int d, void *buf, size_t nbytes) {return -1;}
 static_inline int close(int d) {return -1;}
-- 
2.42.0

